### PR TITLE
feat: add headless renderer example

### PR DESCRIPTION
**Pull request by cs50victor**

### CODE REVIEW

File: examples/headless.rs [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    
+                        let img = match img_bytes.clone().try_into_dynamic() {
+                            Ok(img) => img.to_rgba8(),
+                            Err(e) => panic!("Failed to create image buffer {e:?}"),
+                        };
+    
+                        let images_path =
+                            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test_images");
+                        let uuid = bevy::utils::Uuid::new_v4();
+                        let image_path = images_path.join(format!("{uuid}.png"));
```
Comment (mosure) [inline examples/headless.rs:338]:
```suggestion

                        let images_dir =
                            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test_images");
                        std::fs::create_dir_all(&images_dir).unwrap();

                        let uuid = bevy::utils::Uuid::new_v4();
                        let image_path = images_dir.join(format!("{uuid}.png"));
```

**Review by mosure — APPROVED** [review]
wow, awesome work!! using this technique in future CI correctness testing will be great.

**Comment by cs50victor** [comment]
thanks 🙏🏾. know how to use a server and receive webhooks in bevy?

**Comment by mosure** [comment]
> thanks. know how to use a server and receive webhooks in bevy / an example I can refer to? ( fairly new to bevy )

for a flow like:
- HTTP client sends render request to bevy server
- bevy server responds with an image

I am not sure there is an out-of-box solution that is implemented as a bevy plugin. It looks like someone has started one here (I don't think it is production-ready):
- https://github.com/FloydATC/bevy_httpserver

for a bevy client <-> bevy server flow, naia would be sufficient: https://github.com/naia-lib/naia/tree/main/adapters/bevy/server/src

I think there is good opportunity to use `bevy_httpserver` as a reference to create a more production-ready `bevy_hyper` plugin: https://github.com/hyperium/hyper

**Comment by cs50victor** [comment]
> > thanks. know how to use a server and receive webhooks in bevy / an example I can refer to? ( fairly new to bevy )
> 
> for a flow like:
> 
> * HTTP client sends render request to bevy server
> * bevy server responds with an image
> 
> I am not sure there is an out-of-box solution that is implemented as a bevy plugin. It looks like someone has started one here (I don't think it is production-ready):
> 
> * https://github.com/FloydATC/bevy_httpserver
> 
> for a bevy client <-> bevy server flow, naia would be sufficient: https://github.com/naia-lib/naia/tree/main/adapters/bevy/server/src
> 
> I think there is good opportunity to use `bevy_httpserver` as a reference to create a more production-ready `bevy_hyper` plugin: https://github.com/hyperium/hyper

I'll test these out and give feedback.

**Comment by cs50victor** [comment]
good to merge @mosure  ?

**Comment by mosure** [comment]
LGTM! send it!

**Comment by cs50victor** [comment]
> > thanks. know how to use a server and receive webhooks in bevy / an example I can refer to? ( fairly new to bevy )
> 
> for a flow like:
> 
> * HTTP client sends render request to bevy server
> * bevy server responds with an image
> 
> I am not sure there is an out-of-box solution that is implemented as a bevy plugin. It looks like someone has started one here (I don't think it is production-ready):
> 
> * https://github.com/FloydATC/bevy_httpserver
> 
> for a bevy client <-> bevy server flow, naia would be sufficient: https://github.com/naia-lib/naia/tree/main/adapters/bevy/server/src
> 
> I think there is good opportunity to use `bevy_httpserver` as a reference to create a more production-ready `bevy_hyper` plugin: https://github.com/hyperium/hyper



@mosure  So i looked at these and came up with a 'hacky' way to spawn a listening server using a shared tokio runtime. Probably not the best way to implement this but it kinda works ( https://github.com/cs50victor/kitt2/blob/main/lkgpt/src/server.rs#L245-L319 ). The 'kitt2' repo is  meant to be an open source Google gemini demo with a little extra features like rendering a Gaussian Head avatar ( https://github.com/cs50victor/gaussian-head ) or something similar to Meta's codec avatars ( https://www.youtube.com/watch?v=So8GdQD0Qyc ) that communicates with the user in real time . I've been pulling my hair out for weeks trying to work with async rust, mainly receiving video & audio frames from livekit (https://github.com/cs50victor/kitt2/blob/main/lkgpt/src/main.rs#L167-L191). At the moment it just doesn't return any data from the receiver. Added you to the repo. Would love to get your perspective on what you think might be causing this issue.

**Pull request merged**